### DISCUSSION TITLE
Issues with Exporting JointJS Graph to SVG - Unexpected Elements

### DISCUSSION THREAD

**Discussion by VADDIJASWANT** [clientIO/joint#2730] [Q&A] [answered]
### Introduction

I’m encountering an issue when exporting a JointJS graph to SVG. Sometimes, the exported SVG contains elements that are not present on the paper or the graph itself. I’m not sure why this is happening or how to resolve it.

I have attached two images for reference:

Original Paper/Graph: This shows what the graph looks like on the paper. 
![Screenshot 2024-08-06 140349](https://github.com/user-attachments/assets/82ef4e08-3d97-4fdf-84da-a596b9dbe54a)

Exported SVG: This shows the SVG file after export, where unexpected elements appear. 
![4 Ammonia Process_8_6_2024, 2_03_54 PM](https://github.com/user-attachments/assets/e750e154-4f5f-45a9-99f7-1c20428a62c3)

If anyone has experienced similar issues or has suggestions on how to troubleshoot this, I would greatly appreciate your help.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment, marked as answer]
The element with id `8447566d-4e8b-4bcb-854e-3ec47e90469d` is located in the negative coordinate `y`.
<img width="2279" alt="image" src="https://github.com/user-attachments/assets/49e6f1df-12b1-48da-8f90-7977a1f1894f">

You can test the graph bounding box before calling SVG export. You should see that `graph.getBBox()` gives you a bbox containing the 2 extra elements at the top.

You're probably just no displaying the contents of the paper with negative coordinates, but they are there (either `Paper` or `PaperScroller` configuration).

Ideally, you make sure not to have elements outside viewport. A quick solution is to use the `area` option of `toSVG()` and specify the area you want to be exported.
